extensions/smallrye-metrics/deployment/src/main/java/io/quarkus/smallrye/metrics/deployment/SmallRyeMetricsProcessor.java

```diff
@@ -148,6 +148,18 @@ void annotationTransformers(BuildProducer<AnnotationsTransformerBuildItem> trans
 
                 for (DotName annotationName : clazz.annotations().keySet()) {
```

Comment:
Minor impl. detail - `clazz.annotations().keySet().contains(GAUGE)` would be probaly more performant than iterating over all keys...

Reply:
Good point, changed that